ParcelInvoiceMaker/ParcelInvoiceMaker/ParcelProcessor.swift

```diff
@@ -2,53 +2,99 @@
 //  ParcelInvoiceMaker - ParcelProcessor.swift
 //  Created by yagom.
 //  Copyright © yagom. All rights reserved.
-// 
+//
 
 import Foundation
 
-class ParcelInformation {
-    let address: String
-    var receiverName: String
-    var receiverMobile: String
-    let deliveryCost: Int
-    private let discount: Discount
-    var discountedCost: Int {
-        switch discount {
-        case .none:
-            return deliveryCost
-        case .vip:
-            return deliveryCost / 5 * 4
-        case .coupon:
-            return deliveryCost / 2
-        }
+struct ParcelInformation {
+    let receiverInformation : ReceiverInformation // 변하니까 var
+    let deliveryCost: DeliveryCost //
+    init(receiverInformation : ReceiverInformation, deliveryCost: Int , discount : Discount) {
+        self.receiverInformation = receiverInformation
+        let discountStrategy = discount.strategy
+        self.deliveryCost = DeliveryCost(deliveryCost: deliveryCost, strategy: discountStrategy)
     }
-
-    init(address: String, receiverName: String, receiverMobile: String, deliveryCost: Int, discount: Discount) {
+}
+/// private 과 var , let을 해줘야하는 기준이 헷갈린다. 처음 설정할때 말고 변할일 없으니까 let 이라고 생각
+struct ReceiverInformation{
+    let address : String
+    let name : String
+    let mobile : String
+    init(address: String, name: String, mobile: String) {
         self.address = address
-        self.receiverName = receiverName
-        self.receiverMobile = receiverMobile
+        self.name = name
+        self.mobile = mobile
+    }
+}
+//MARK: - Discount
+struct DeliveryCost {
+    let deliveryCost : Int
+    var discountStrategy : DiscountStrategy
+    var discountedCost : Int {
+        return discountStrategy.applyDiscount(deliveryCost: deliveryCost)
+    }
+    init(deliveryCost: Int, strategy: DiscountStrategy) {
         self.deliveryCost = deliveryCost
-        self.discount = discount
+        self.discountStrategy = strategy
+    }
+}
+protocol DiscountStrategy {
+    func applyDiscount(deliveryCost : Int) -> Int
+}
+
+final class NoDiscount : DiscountStrategy{
```

Comment:
```suggestion
final class NoDiscount: DiscountStrategy {
```
: 앞에는 붙여주시고, { 앞에는 띄워주세요!